reference/lifecycle-terminology.md

```diff
@@ -0,0 +1,61 @@
+Software Lifecycle Terminology
+==============================
+
+We work on software that's anywhere in its lifecycle - from something that we've created just to see if it's possible, through to widely-used and well-maintained software that's used in production. 
+
+In order to be clear about what people should expect from a piece of software, we use specific terminology. This page documents our interpretation of these terms, and is intended to be referred to by developers at ODSC, developers that we're collaborating with, and clients. 
+
+Any particular piece of software might have elements that are at different stages of development. For example, a new feature in a mature software product might be a prototype, while the main software product is production-ready. 
+
+As a rule of thumb, it's approximately an order of magnitude of extra work, complexity or cost to move up each level; that is to say that a production implementation of a feature is 100x more complex than its prototype. 
+
+## Prototype
+
+A prototype has been created to see if something is technically feasible, or to demonstrate what something could look like. 
+
+It will:
+* Be quick to make
+* Serve a single purpose - for example to try out an interface idea, or to test an algorithm
+* Provide some learning for a future implementation of a feature. 
+
+
+It won't:
+* Deal with any cases that it wasn't explicitly designed for
```

Comment:
No software can deal with a case that it wasn't explicitly designed for ;) I think I get what you're saying though, more like "won't cover all corner cases" or be very limited in use cases implemented, something along that line?

Reply:
In practical terms, I'd be ok with a prototype crashing the web server when an invalid spreadsheet was uploaded, whereas some software at MVP level or above I'd expect to fail less catastrophically. 

Perhaps this would be a good example to use throughout the docs:
- prototype: it's ok if invalid input results in catastrophic failure
- MVP: anything other than valid input should result in a simple error (eg "I don't know what to do with this") 
- Production: invalid input should have been thought about and analysis should be carried out to try to help the user